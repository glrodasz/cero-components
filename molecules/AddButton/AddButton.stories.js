import { AddButton, options, styles } from '.'
import {
  getTemplate,
  getListTemplate,
  getOptionsArgTypes,
} from '../../helpers/storybook'

const Template = getTemplate(AddButton, styles)
const ListTemplate = getListTemplate(AddButton, styles)

export default {
  title: 'Molecules/AddButton',
  component: AddButton,
  args: {
    children: "Caves they've pick stealth Dwarf?",
  },
  argTypes: {
    type: getOptionsArgTypes(options.types),
    icon: getOptionsArgTypes(options.icons),
    onAdd: { defaultValue: null },
  },
}

export const Default = Template.bind({})

export const Types = ListTemplate.bind({})
Types.args = { items: options.types.map((type) => ({ type })) }

export const EditMode = Template.bind({})
EditMode.args = {
  defaultValue: 'Fixed caught squash wax Silvan sweet-taking keeps.',
  defaultEditMode: true,
}

export const HelpText = Template.bind({})
HelpText.args = {
  defaultValue: "Understand bravely halfling's rank balls.",
  defaultEditMode: true,
  focusHelpText: 'Fly, you fools!',
  blurHelpText: "Morgul-rats cripples you'll?",
}
