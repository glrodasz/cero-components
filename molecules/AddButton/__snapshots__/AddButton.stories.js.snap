// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Storyshots Molecules/AddButton Default 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="add-button type-primary"
    onClick={[Function]}
  >
    <div
      className="icon color-base size-md background-highlight"
      style={
        Object {
          "height": 35,
          "width": 35,
        }
      }
    >
      <svg
        height={35}
        viewBox="0 0 20 20"
        width={35}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M10 20C4.48612 20 0 15.5139 0 10C0 4.48612 4.48612 0 10 0C15.5139 0 20 4.48612 20 10C20 15.5139 15.5139 20 10 20ZM10 1.09864C5.09155 1.09864 1.09864 5.09155 1.09864 10C1.09864 14.9084 5.09155 18.9014 10 18.9014C14.9084 18.9014 18.9014 14.9084 18.9014 10C18.9014 5.09155 14.9084 1.09864 10 1.09864Z"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M4.68396 9.4093H15.316V10.5906H4.68396V9.4093Z"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M9.40942 4.68396H10.5908V15.316H9.40942V4.68396Z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div
      className="spacer vertical"
      style={
        Object {
          "height": "100%",
          "maxHeight": "auto",
          "width": 15,
        }
      }
    />
    <div
      className="paragraph color-base size-md weight-medium"
    >
      Caves they've pick stealth Dwarf?
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/AddButton Edit Mode 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="add-button type-primary is-editable"
    onClick={[Function]}
  >
    <div
      className="edit-container"
    >
      <input
        autoFocus={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        type="text"
        value="Fixed caught squash wax Silvan sweet-taking keeps."
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/AddButton Help Text 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="add-button type-primary is-editable"
    onClick={[Function]}
  >
    <div
      className="edit-container"
    >
      <input
        autoFocus={false}
        onBlur={[Function]}
        onChange={[Function]}
        onFocus={[Function]}
        onKeyDown={[Function]}
        type="text"
        value="Understand bravely halfling's rank balls."
      />
      <div
        className="help-text paragraph color-inverted size-sm weight-normal"
      >
        Morgul-rats cripples you'll?
      </div>
      <div
        className="spacer vertical"
        style={
          Object {
            "height": "100%",
            "maxHeight": "auto",
            "width": 15,
          }
        }
      />
    </div>
  </div>
</div>
`;

exports[`Storyshots Molecules/AddButton Types 1`] = `
<div
  style={
    Object {
      "alignContent": "flex-start",
      "display": "flex",
      "flexDirection": "column",
      "flexWrap": "wrap",
      "gap": "10px 30px",
      "height": "100%",
      "justifyContent": "flex-start",
      "maxHeight": "auto",
    }
  }
>
  <div
    className="add-button type-primary"
    onClick={[Function]}
  >
    <div
      className="icon color-base size-md background-highlight"
      style={
        Object {
          "height": 35,
          "width": 35,
        }
      }
    >
      <svg
        height={35}
        viewBox="0 0 20 20"
        width={35}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M10 20C4.48612 20 0 15.5139 0 10C0 4.48612 4.48612 0 10 0C15.5139 0 20 4.48612 20 10C20 15.5139 15.5139 20 10 20ZM10 1.09864C5.09155 1.09864 1.09864 5.09155 1.09864 10C1.09864 14.9084 5.09155 18.9014 10 18.9014C14.9084 18.9014 18.9014 14.9084 18.9014 10C18.9014 5.09155 14.9084 1.09864 10 1.09864Z"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M4.68396 9.4093H15.316V10.5906H4.68396V9.4093Z"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M9.40942 4.68396H10.5908V15.316H9.40942V4.68396Z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div
      className="spacer vertical"
      style={
        Object {
          "height": "100%",
          "maxHeight": "auto",
          "width": 15,
        }
      }
    />
    <div
      className="paragraph color-base size-md weight-medium"
    >
      Caves they've pick stealth Dwarf?
    </div>
  </div>
  <div
    className="add-button type-secondary"
    onClick={[Function]}
  >
    <div
      className="icon color-highlight size-md"
      style={
        Object {
          "height": 35,
          "width": 35,
        }
      }
    >
      <svg
        height={35}
        viewBox="0 0 20 20"
        width={35}
        xmlns="http://www.w3.org/2000/svg"
      >
        <path
          clipRule="evenodd"
          d="M10 20C4.48612 20 0 15.5139 0 10C0 4.48612 4.48612 0 10 0C15.5139 0 20 4.48612 20 10C20 15.5139 15.5139 20 10 20ZM10 1.09864C5.09155 1.09864 1.09864 5.09155 1.09864 10C1.09864 14.9084 5.09155 18.9014 10 18.9014C14.9084 18.9014 18.9014 14.9084 18.9014 10C18.9014 5.09155 14.9084 1.09864 10 1.09864Z"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M4.68396 9.4093H15.316V10.5906H4.68396V9.4093Z"
          fillRule="evenodd"
        />
        <path
          clipRule="evenodd"
          d="M9.40942 4.68396H10.5908V15.316H9.40942V4.68396Z"
          fillRule="evenodd"
        />
      </svg>
    </div>
    <div
      className="spacer vertical"
      style={
        Object {
          "height": "100%",
          "maxHeight": "auto",
          "width": 15,
        }
      }
    />
    <div
      className="paragraph color-base size-md weight-medium"
    >
      Caves they've pick stealth Dwarf?
    </div>
  </div>
</div>
`;
